packages/core/src/shared/extensions/ssh.ts

```diff
@@ -119,6 +123,26 @@ export class RemoteSshSettings extends Settings.define('remote.SSH', remoteSshTy
     }
 }
 
+export async function testSshConnection(
+    ProcessClass: typeof ChildProcess,
+    hostname: string,
+    sshPath: string,
+    user: string,
+    session: SSM.StartSessionResponse
+): Promise<void> {
+    try {
+        const env = { SESSION_ID: session.SessionId, STREAM_URL: session.StreamUrl, TOKEN: session.TokenValue }
+        await new ProcessClass(sshPath, ['-T', `${user}@${hostname}`, 'echo connected && exit']).run({
```

Comment:
can we capture the output of this process invocation and include it in the logs? 

we may even want to use a similar approach as `collectSamErrors` to collect errors, and include them in the user-facing message. https://github.com/aws/aws-toolkit-vscode/blob/3d46b7ba38fb86f6ce0ec082fe648b9167f12f1d/packages/core/src/shared/sam/cli/samCliInvokerUtils.ts#L75

Reply:
I believe the output is already in the logs because we bind a logger to the process: 
https://github.com/aws/aws-toolkit-vscode/blob/823a0e3f0d0cdb416bbd8f27ecd2fe7fc1d73288/packages/core/src/awsService/ec2/model.ts#L248-L252
Do you think its worth filtering the logs? I agree that `collectSamErrors` method could definitely be generalized and potentially applied here as a filter, but unsure what to filter for.